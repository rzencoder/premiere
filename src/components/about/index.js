import React from "react";
import { Container, Item, Divider, Text, Title, Icon } from "./styles/about";

export default function About({ children, ...restProps }) {
  return <Container {...restProps}>{children}</Container>;
}

About.Item = function AboutItem({ children, ...restProps }) {
  return <Item {...restProps}>{children}</Item>;
};

About.Divider = function AboutDivider({ children, ...restProps }) {
  return <Divider {...restProps} />;
};

About.Title = function AboutTitle({ children, ...restProps }) {
  return <Title {...restProps}>{children}</Title>;
};

About.Text = function AboutText({ children, ...restProps }) {
  return <Text {...restProps}>{children}</Text>;
};

About.Icon = function AboutIcon({ src, ...restProps }) {
  return <Icon {...restProps} src={src} />;
};
