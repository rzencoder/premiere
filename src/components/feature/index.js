import React from "react";
import {
  Background,
  Container,
  Overlay,
  TextContainer,
  Title,
  Text,
  Button,
  Details,
  SubText,
  Rating,
  AgeRating,
  Year,
} from "./styles/feature";

export default function Feature({ children, ...restProps }) {
  return <Container {...restProps}>{children}</Container>;
}

Feature.Background = function FeatureBackground({ ...restProps }) {
  return <Background {...restProps} />;
};

Feature.Overlay = function FeatureOverlay({ ...restProps }) {
  return <Overlay {...restProps} />;
};

Feature.TextContainer = function FeatureTextContainer({
  children,
  ...restProps
}) {
  return <TextContainer {...restProps}>{children}</TextContainer>;
};

Feature.Title = function FeatureTitle({ children, ...restProps }) {
  return <Title {...restProps}>{children}</Title>;
};

Feature.Text = function FeatureText({ children, ...restProps }) {
  return <Text {...restProps}>{children}</Text>;
};

Feature.Button = function FeatureButton({ children, ...restProps }) {
  return <Button {...restProps}>{children}</Button>;
};

Feature.Details = function FeatureDetails({ children, ...restProps }) {
  return <Details {...restProps}>{children}</Details>;
};

Feature.SubText = function FeatureSubText({ children, ...restProps }) {
  return <SubText {...restProps}>{children}</SubText>;
};

Feature.AgeRating = function FeatureAgeRating({ ...restProps }) {
  return <AgeRating {...restProps} />;
};

Feature.Year = function FeatureYear({ children, ...restProps }) {
  return (
    <Year {...restProps}>
      <div>{children}</div>
    </Year>
  );
};

Feature.Rating = function FeatureRating({ children, ...restProps }) {
  return (
    <Rating {...restProps}>
      <div>{children}</div>
    </Rating>
  );
};
