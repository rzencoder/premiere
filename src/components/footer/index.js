import React from "react";
import {
  Container,
  Text,
  Subtitle,
  Section,
  Link,
  Divider,
  Column,
  Break,
} from "./styles/footer";

export default function Footer({ children, ...restProps }) {
  return <Container {...restProps}>{children}</Container>;
}

Footer.Text = function FooterText({ children, ...restProps }) {
  return <Text {...restProps}>{children}</Text>;
};

Footer.Subtitle = function FooterSubtitle({ children, ...restProps }) {
  return <Subtitle {...restProps}>{children}</Subtitle>;
};

Footer.Link = function FooterLink({ children, ...restProps }) {
  return <Link {...restProps}>{children}</Link>;
};

Footer.Divider = function FooterDivider({ ...restProps }) {
  return <Divider {...restProps} />;
};

Footer.Section = function FooterSection({ children, ...restProps }) {
  return <Section {...restProps}>{children}</Section>;
};

Footer.Column = function FooterColumn({ children, ...restProps }) {
  return <Column {...restProps}>{children}</Column>;
};

Footer.Break = function FooterBreak({ ...restProps }) {
  return <Break {...restProps} />;
};
