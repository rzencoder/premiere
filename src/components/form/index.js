import React from "react";
import { Link as ReactRouterLink } from "react-router-dom";
import {
  Container,
  Title,
  Content,
  Label,
  Input,
  Submit,
  Frame,
  Text,
  Link,
  Error,
} from "./styles/form";

export default function Form({ children, ...restProps }) {
  return <Container {...restProps}>{children}</Container>;
}

Form.Frame = function FormFrame({ children, ...restProps }) {
  return (
    <Frame {...restProps}>
      <div>{children}</div>
    </Frame>
  );
};

Form.Title = function FormTitle({ children, ...restProps }) {
  return <Title {...restProps}>{children}</Title>;
};

Form.Content = function FormContent({ children, ...restProps }) {
  return <Content {...restProps}>{children}</Content>;
};

Form.Label = function FormLabel({ children, ...restProps }) {
  return <Label {...restProps}>{children}</Label>;
};

Form.Input = function FormInput({ ...restProps }) {
  return <Input {...restProps} />;
};

Form.Submit = function FormSubmit({ children, ...restProps }) {
  return <Submit {...restProps}>{children}</Submit>;
};

Form.Text = function FormText({ children, ...restProps }) {
  return <Text {...restProps}>{children}</Text>;
};

Form.Link = function FormLink({ to, children, ...restProps }) {
  return (
    <ReactRouterLink to={to}>
      <Link {...restProps}>{children}</Link>
    </ReactRouterLink>
  );
};

Form.Error = function FormError({ children, ...restProps }) {
  return <Error {...restProps}>{children}</Error>;
};
