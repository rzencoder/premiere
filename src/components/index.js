export { default as Navigation } from "./navigation";
export { default as Jumbotron } from "./jumbotron";
export { default as About } from "./about";
export { default as Footer } from "./footer";
export { default as Form } from "./form";
