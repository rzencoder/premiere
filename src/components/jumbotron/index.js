import React from "react";
import {
  Background,
  Container,
  Overlay,
  Group,
  Title,
  Text,
  CallToAction,
} from "./styles/jumbotron";
import { Link as ReactRouterLink } from "react-router-dom";

export default function Jumbotron({ children, ...restProps }) {
  return <Container {...restProps}>{children}</Container>;
}

Jumbotron.Overlay = function JumbotronOverlay({ ...restProps }) {
  return <Overlay {...restProps} />;
};

Jumbotron.Group = function JumbotronGroup({ children, ...restProps }) {
  return <Group {...restProps}>{children}</Group>;
};
Jumbotron.Title = function JumbotronTitle({ children, ...restProps }) {
  return <Title {...restProps}>{children}</Title>;
};

Jumbotron.Text = function JumbotronText({ children, ...restProps }) {
  return <Text {...restProps}>{children}</Text>;
};

Jumbotron.CallToAction = function JumbotronCallToAction({
  to,
  children,
  ...restProps
}) {
  return (
    <ReactRouterLink to={to}>
      <CallToAction {...restProps}>{children}</CallToAction>
    </ReactRouterLink>
  );
};

Jumbotron.Background = function JumbotronBackground({ ...restProps }) {
  return <Background {...restProps} />;
};
