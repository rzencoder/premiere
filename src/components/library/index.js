import React from "react";
import {
  Container,
  Image,
  Button,
  Divider,
  CarouselContainer,
  CarouselGroup,
  CarouselButton,
  Selector,
  SelectorItem,
  CarouselOverlay,
  Panel,
} from "./styles/library";

export default function Library({ children, ...restProps }) {
  return <Container {...restProps}>{children}</Container>;
}

Library.Button = function LibraryButton({ children, ...restProps }) {
  return <Button {...restProps}>{children}</Button>;
};

Library.Divider = function LibraryDivider({ ...restProps }) {
  return <Divider {...restProps} />;
};

Library.Image = function LibraryImage({ ...restProps }) {
  return <Image {...restProps} />;
};

Library.CarouselContainer = function LibraryCarouselContainer({
  children,
  ...restProps
}) {
  return <CarouselContainer {...restProps}>{children}</CarouselContainer>;
};

Library.CarouselGroup = function LibraryCarouselGroup({
  children,
  ...restProps
}) {
  return <CarouselGroup {...restProps}>{children}</CarouselGroup>;
};

Library.CarouselButton = function LibraryCarouselButton({
  children,
  ...restProps
}) {
  return <CarouselButton {...restProps}>{children}</CarouselButton>;
};

Library.Selector = function LibrarySelector({ children, ...restProps }) {
  return (
    <Selector {...restProps}>
      <div>{children}</div>
    </Selector>
  );
};

Library.SelectorItem = function LibrarySelectorItem({
  children,
  ...restProps
}) {
  return <SelectorItem {...restProps}>{children}</SelectorItem>;
};

Library.CarouselOverlay = function LibraryCarouselOverlay({ ...restProps }) {
  return <CarouselOverlay {...restProps} />;
};

Library.Panel = function LibraryPanel({ children, ...restProps }) {
  return (
    <Panel {...restProps}>
      <div>{children}</div>
    </Panel>
  );
};
