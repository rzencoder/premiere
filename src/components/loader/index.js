import React from "react";
import { Container, Spinner } from "./styles/loader";

export default function Loader({ ...restProps }) {
  return (
    <Container data-testid="loader">
      <Spinner {...restProps} />;
    </Container>
  );
}
