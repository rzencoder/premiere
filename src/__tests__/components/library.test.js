import React from "react";
import { render, fireEvent } from "@testing-library/react";
import { Library } from "../../components";
import Carousel from "nuka-carousel";

const genres = ["Action", "Comedy", "Drama"];

const content = [
  {
    title: "Blade Runner 2049",
    genre: ["Action", "Drama", "Mystery"],
    slug: "blade-runner-2049",
  },
  {
    title: "Inception",
    genre: ["Action", "Sci Fi"],
    slug: "inception",
  },
  {
    title: "500 Days of Summer",
    genre: ["Comedy", "Romance", "Drama"],
    slug: "days-of-summer",
  },
  {
    title: "Sunshine",
    genre: ["Sci Fi"],
    slug: "sunshine",
  },
  {
    title: "Amélie",
    genre: ["Comedy", "Romance"],
    slug: "amelie",
  },
];

describe("<Library />", () => {
  it("renders <Library /> component", () => {
    const { container, getByText, getByAltText, getByTestId } = render(
      <>
        <Library.Divider />
        <Library>
          <Library.Selector>
            {genres.map((item) => (
              <Library.SelectorItem active={true} key={item} onClick={() => {}}>
                {item}
              </Library.SelectorItem>
            ))}
          </Library.Selector>
          <Library.CarouselContainer>
            <Library.Panel>Streaming on Premiere</Library.Panel>
            <Library.CarouselGroup>
              <Carousel
                slidesToShow={4}
                wrapAround={true}
                withoutControls={false}
                renderBottomCenterControls={null}
                renderCenterLeftControls={({ previousSlide }) => (
                  <Library.CarouselButton onClick={previousSlide}>
                    {"<"}
                  </Library.CarouselButton>
                )}
                renderCenterRightControls={({ nextSlide }) => (
                  <Library.CarouselButton onClick={nextSlide}>
                    {">"}
                  </Library.CarouselButton>
                )}
              >
                {content.map((item) => {
                  return (
                    <Library.Image
                      src={`../images/content/${item.slug}-poster.jpg`}
                      alt={item.title}
                      key={item.slug}
                      onClick={() => {}}
                      onKeyDown={() => {}}
                    />
                  );
                })}
              </Carousel>
              <Library.CarouselOverlay show={true} />
            </Library.CarouselGroup>
          </Library.CarouselContainer>
        </Library>
      </>
    );

    expect(getByText("Action")).toBeTruthy();
    expect(getByText("Comedy")).toBeTruthy();
    expect(getByText("Drama")).toBeTruthy();
    expect(getByAltText("Blade Runner 2049")).toBeTruthy();
    expect(getByAltText("Inception")).toBeTruthy();
    expect(getByAltText("500 Days of Summer")).toBeTruthy();
    expect(getByAltText("Sunshine")).toBeTruthy();
    expect(getByAltText("Amélie")).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });

  it("shows/hides items when carousel button is pressed", () => {
    const { container, getByText, getByAltText, getByTestId } = render(
      <Library>
        <Library.CarouselContainer>
          <Library.Panel>Streaming on Premiere</Library.Panel>
          <Library.CarouselGroup>
            <Carousel
              slidesToShow={4}
              wrapAround={true}
              withoutControls={false}
              renderBottomCenterControls={null}
              renderCenterLeftControls={({ previousSlide }) => (
                <Library.CarouselButton onClick={previousSlide}>
                  {"<"}
                </Library.CarouselButton>
              )}
              renderCenterRightControls={({ nextSlide }) => (
                <Library.CarouselButton
                  data-testid="carousel-right"
                  onClick={nextSlide}
                >
                  {">"}
                </Library.CarouselButton>
              )}
            >
              {content.map((item) => {
                return (
                  <Library.Image
                    src={`../images/content/${item.slug}-poster.jpg`}
                    alt={item.title}
                    key={item.slug}
                    onClick={() => {}}
                    onKeyDown={() => {}}
                  />
                );
              })}
            </Carousel>
            <Library.CarouselOverlay show={true} />
          </Library.CarouselGroup>
        </Library.CarouselContainer>
      </Library>
    );

    expect(getByAltText("Blade Runner 2049")).toBeTruthy();
    expect(getByAltText("Amélie")).toBeTruthy();
    expect(getByAltText("Amélie").getAttribute("aria-hidden")).toEqual("true");
    expect(
      getByAltText("Blade Runner 2049").getAttribute("aria-hidden")
    ).toEqual("false");
    fireEvent.click(getByTestId("carousel-right"));
    expect(getByAltText("Amélie").getAttribute("aria-hidden")).toEqual("false");
    expect(
      getByAltText("Blade Runner 2049").getAttribute("aria-hidden")
    ).toEqual("true");
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders a different style to unselected genres", () => {
    const { container, getByText } = render(
      <Library>
        <Library.Selector>
          {genres.map((item) => (
            <Library.SelectorItem active={false} key={item} onClick={() => {}}>
              {item}
            </Library.SelectorItem>
          ))}
        </Library.Selector>
      </Library>
    );

    expect(getByText("Action")).toBeTruthy();
    expect(getByText("Comedy")).toBeTruthy();
    expect(getByText("Drama")).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });

  it("hides overlay when show is false", () => {
    const { container, getByText } = render(
      <Library>
        <Library.CarouselContainer>
          <Library.Panel>Streaming on Premiere</Library.Panel>
          <Library.CarouselGroup>
            <Library.CarouselOverlay show={false} />
          </Library.CarouselGroup>
        </Library.CarouselContainer>
      </Library>
    );

    expect(getByText("Streaming on Premiere")).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });
});
