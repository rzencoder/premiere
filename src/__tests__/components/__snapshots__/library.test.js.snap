// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Library /> hides overlay when show is false 1`] = `
<section
  class="library__Container-meah2n-0 laSfGJ"
>
  <div
    class="library__CarouselContainer-meah2n-3 hHyGPX"
  >
    <div
      class="library__Panel-meah2n-9 deXvZh"
    >
      <div>
        Streaming on Premiere
      </div>
    </div>
    <div
      class="library__CarouselGroup-meah2n-4 fEGrxX"
    >
      <div
        class="library__CarouselOverlay-meah2n-8 gfrAkp"
      />
    </div>
  </div>
</section>
`;

exports[`<Library /> renders <Library /> component 1`] = `
<div
  class="library__Divider-meah2n-2 labXmN"
/>
`;

exports[`<Library /> renders a different style to unselected genres 1`] = `
<section
  class="library__Container-meah2n-0 laSfGJ"
>
  <div
    class="library__Selector-meah2n-6 bhKIRv"
  >
    <div>
      <button
        class="library__SelectorItem-meah2n-7 fjYCKl"
      >
        Action
      </button>
      <button
        class="library__SelectorItem-meah2n-7 fjYCKl"
      >
        Comedy
      </button>
      <button
        class="library__SelectorItem-meah2n-7 fjYCKl"
      >
        Drama
      </button>
    </div>
  </div>
</section>
`;

exports[`<Library /> shows/hides items when carousel button is pressed 1`] = `
<section
  class="library__Container-meah2n-0 laSfGJ"
>
  <div
    class="library__CarouselContainer-meah2n-3 hHyGPX"
  >
    <div
      class="library__Panel-meah2n-9 deXvZh"
    >
      <div>
        Streaming on Premiere
      </div>
    </div>
    <div
      class="library__CarouselGroup-meah2n-4 fEGrxX"
    >
      <div
        class="slider"
        style="box-sizing: border-box; display: block; height: inherit; position: relative; width: 100%;"
        tabindex="0"
      >
        <div
          aria-atomic="true"
          aria-live="polite"
          style="position: absolute; left: -10000px; width: 1px; height: 1px; overflow: hidden;"
          tabindex="-1"
        >
          Slide 2 of 5
        </div>
        <div
          class="slider-frame"
          style="box-sizing: border-box; display: block; height: 100%; margin: 0px; overflow: hidden; padding: 0px; position: relative; transform: translate3d(0, 0, 0);"
        >
          <ul
            class="slider-list"
            style="box-sizing: border-box; cursor: pointer; display: block; height: 0px; margin: 0px 0px; padding: 0px; position: relative; transform: translate3d(0px, 0px, 0); width: auto;"
          >
            <li
              class="slider-slide"
              inert="true"
              style="box-sizing: border-box; display: inline-block; height: auto; left: 0px; list-style-type: none; margin: auto 0px auto 0px; position: absolute; top: 0px; transform: scale(1); transition: transform .4s linear; vertical-align: top; width: 0px;"
              tabindex="-1"
            >
              <img
                alt="Blade Runner 2049"
                aria-hidden="true"
                class="library__Image-meah2n-1 dgRpm"
                src="../images/content/blade-runner-2049-poster.jpg"
              />
            </li>
            <li
              class="slider-slide slide-visible slide-current"
              style="box-sizing: border-box; display: inline-block; height: auto; left: 0px; list-style-type: none; margin: auto 0px auto 0px; position: absolute; top: 0px; transform: scale(1); transition: transform .4s linear; vertical-align: top; width: 0px;"
              tabindex="-1"
            >
              <img
                alt="Inception"
                aria-hidden="false"
                class="library__Image-meah2n-1 dgRpm"
                src="../images/content/inception-poster.jpg"
                tabindex="0"
              />
            </li>
            <li
              class="slider-slide slide-visible"
              inert="true"
              style="box-sizing: border-box; display: inline-block; height: auto; left: 0px; list-style-type: none; margin: auto 0px auto 0px; position: absolute; top: 0px; transform: scale(1); transition: transform .4s linear; vertical-align: top; width: 0px;"
              tabindex="-1"
            >
              <img
                alt="500 Days of Summer"
                aria-hidden="false"
                class="library__Image-meah2n-1 dgRpm"
                src="../images/content/days-of-summer-poster.jpg"
                tabindex="0"
              />
            </li>
            <li
              class="slider-slide slide-visible"
              inert="true"
              style="box-sizing: border-box; display: inline-block; height: auto; left: 0px; list-style-type: none; margin: auto 0px auto 0px; position: absolute; top: 0px; transform: scale(1); transition: transform .4s linear; vertical-align: top; width: 0px;"
              tabindex="-1"
            >
              <img
                alt="Sunshine"
                aria-hidden="false"
                class="library__Image-meah2n-1 dgRpm"
                src="../images/content/sunshine-poster.jpg"
                tabindex="0"
              />
            </li>
            <li
              class="slider-slide slide-visible"
              inert="true"
              style="box-sizing: border-box; display: inline-block; height: auto; left: 0px; list-style-type: none; margin: auto 0px auto 0px; position: absolute; top: 0px; transform: scale(1); transition: transform .4s linear; vertical-align: top; width: 0px;"
              tabindex="-1"
            >
              <img
                alt="AmÃ©lie"
                aria-hidden="false"
                class="library__Image-meah2n-1 dgRpm"
                src="../images/content/amelie-poster.jpg"
                tabindex="0"
              />
            </li>
          </ul>
        </div>
        <div
          class="slider-control-centerleft"
          style="position: absolute; top: 50%; left: 0px; transform: translateY(-50%);"
        >
          <button
            class="library__CarouselButton-meah2n-5 eZTVsX"
          >
            &lt;
          </button>
        </div>
        <div
          class="slider-control-centerright"
          style="position: absolute; top: 50%; right: 0px; transform: translateY(-50%);"
        >
          <button
            class="library__CarouselButton-meah2n-5 eZTVsX"
            data-testid="carousel-right"
          >
            &gt;
          </button>
        </div>
        <style
          type="text/css"
        >
          .slider-slide &gt; img { width: 100%; display: block; }
          .slider-slide &gt; img:focus { margin: auto; }
        </style>
      </div>
      <div
        class="library__CarouselOverlay-meah2n-8 dEQzzq"
      />
    </div>
  </div>
</section>
`;
